### ISSUE TITLE
Black silently changes expression type when wrapping expressions in parenthesis

### ISSUE THREAD

**Issue by andrewv99** [psf/black#2366]
**Describe the bug**

Black sometimes wraps an expression that is already long (enough), or possibly silently becomes long (enough) as a result of refactoring across a codebase and then applying Black formatting to it. The result is that the expression becomes a tuple of length 1, which is not the same as a scalar, and that change in the code by Black then silently causes type sensitive bugs to manifest that were not there before.
[BlackTrailingCommaBug.zip](https://github.com/psf/black/files/6790519/BlackTrailingCommaBug.zip)

**To Reproduce**

Run the small sample code inside the attached zip, which as been Black formatted.
See detailed comments in the sample code which explain the issue and possible fixes.

**Expected behavior**

The behaviour of code must not change with Black formatting

**Environment:**

$ black --version
black, version 21.6b0
$ python --version
Python 3.6.8
$ cat /etc/os-release |grep PRETTY_NAME
PRETTY_NAME="Ubuntu 18.04.2 LTS"

**Does this bug also happen on main?**

Yes, tried on <https://black.vercel.app/?version=main>

**Additional context**

None

**Comment by felix-hilden** [comment]
Hi, this question is very similar to #2267. Adding a set of parenthesis does not change the type of that expression. Commas do, but I don't think any are added to expressions that are only parenthesised. I can't reproduce the case you presented in the file using the [playground](https://black.vercel.app/?version=main&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AIAANVdAD2IimZxl1N_WlXnON2nzNG_BfI5IiHRNS5nuiFUq2Gw1xY_r1fOHlVioCRjnWjYyFs1mvprlgSK8X8E4v_fiBx_l4Su-V4GAlOZbS1tjovYpZ03nGSioORHkuBhB_hRZePIdP_fN_lNiAw1nerVzmOmWNIsXGh35VS5VMCUpaSaj1xQlA9diNGX9zBOYUNGykjBcOVIAl3j9OHNIOu6OUP8nwNV6P-sksL4O1TZ9b034MxCWvxPr-JNyUU1x2JE8x8tCIU1d_IMIgutOCi5-eOTt80Q-QAAAACk-bIAQMS65wAB8QGBBAAAU04XcrHEZ_sCAAAAAARZWg==). Note the missing comma on line 13 of the result. Could you provide a buggy example there and link to it?

**Comment by andrewv99** [comment]
Sorry, you are right, it is the comma that changes the type of the expression to a tuple. What I had experienced was that Black decided to wrap the expression in parenthesis (when it got sufficiently long I assume) but then *also add* a comma - however, I need to see if I can reproduce that case first - I will try to do that - I thought I had, but no. I extracted and changed this sample code from a much larger code base, as one example where I am pretty sure Black reformatting had added the comma along with the parenthesis (i.e. I am pretty sure there was no comma before formatting). Basically, the large code base was working ok - then I decided to Black format it, and then I got failures, and tracked the cause down a runtime error triggered by the type being a tuple, when the larger code base used to work fine (and did rely on the expressions in question being scalar).

**Comment by felix-hilden** [comment]
Fair enough, thanks for digging some more! It's not that I don't believe you, it's just hard to do anything without concrete examples of failure. I hope you find the problem or are able to conclude it was some other error 👍 Honestly, just formatting the code base again might give a clue!

**Comment by andrewv99** [comment]
Yes, a reproducible error is needed for sure. Will try to create one or find anything that can discount my experience - either way. Thanks for the product and your support :)

**Comment by ichard26** [comment]
Hey @andrewv99, was a reproducible case ever found? It's been quite a while so I wanted to check in :)

**Comment by santriseus** [comment]
Still reproduced in current version of VSCode pulgin (v2024.2.0)
```
     image_info = (
          "yaw_left.png", ("Rotate camera left", (36, 255, 12))
          ) if yaw_delta_angle < 0 else (
          "yaw_right.png", ("Rotate camera right", (36, 255, 12))
      )
```
Becomes
```
        image_info = (
            ("yaw_left.png", ("Rotate camera left", (36, 255, 12)))
            if yaw_delta_angle < 0
            else (
                "yaw_right.png",
                ("Rotate camera right", (36, 255, 12)),
            )
        )
```

**Comment by JelleZijlstra** [comment]
Your example doesn't show that Black changed any expression types.
